src/main/java/net/minecraftforge/items/IItemHandler.java

```diff
@@ -19,73 +19,262 @@
 
 package net.minecraftforge.items;
 
+import com.google.common.collect.Range;
 import net.minecraft.item.ItemStack;
+import net.minecraft.util.NonNullList;
+import net.minecraftforge.items.filter.IStackFilter;
+import net.minecraftforge.items.filter.ItemStackFilter;
 
 import javax.annotation.Nonnull;
 
 public interface IItemHandler
```

Comment:
You should make this a list so that we can iterate over slots with a list iterator

Reply:
does the IItemHandlerIterator work for you ?